### DISCUSSION TITLE
How to make ExecutableReturnsRows from text

### DISCUSSION THREAD

**Discussion by reubengann** [sqlalchemy/sqlalchemy#9395] [Usage Questions] [answered]
I just migrated from 1.4, and this statement

`query = select(ModelA).from_statement(text(stmt_1))`

now generates a type hint warning

Argument of type "TextClause" cannot be assigned to parameter "statement" of type "ExecutableReturnsRows" in function "from_statement"
  "TextClause" is incompatible with "ExecutableReturnsRows"

How can I make a valid ExecutableReturnsRows statement from a raw SQL query?

**Comment by zzzeek** [comment, marked as answer]
you'll want to call [.columns()](https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#getting-orm-results-from-textual-statements) on that text() construct so that the return types are known, which will also give you the statement form you're looking for.

**Reply by reubengann** [reply]
Thanks, but now I get

Argument of type "TextualSelect" cannot be assigned to parameter "statement" of type "ExecutableReturnsRows" in function "from_statement"

**Reply by zzzeek** [reply]
that's a bug.  wait for version 2.0.5 please  #9398
